KenticoOnboardingApplication/test/KenticoOnboardingApplication.Tests.Base/ArgWrapper.cs

```diff
@@ -0,0 +1,14 @@
+﻿using KenticoOnboardingApplication.Contracts.Models;
+using KenticoOnboardingApplication.Tests.Base.Comparers;
+using NSubstitute;
+
+namespace KenticoOnboardingApplication.Tests.Base
+{
+    public static class ArgWrapper
+    {
+        private static readonly ItemComparer s_comparer = new ItemComparer();
```

Comment:
It is not [Lazy™](https://docs.microsoft.com/en-us/dotnet/api/system.lazy-1?view=netframework-4.7.2) while the one in the `EqualConstraintExtensions` is, but the other makes this one one-too-many (or vice versa, whichever you love most 😛). 

What about moving the lazy static property from `EqualConstraintExtensions` into the `ItemComparer` and make its constructor private so the lazy instance is the only one available in the eternity of the solution 😃